tycho-targetplatform/src/main/java/org/eclipse/tycho/targetplatform/TargetDefinitionFile.java

```diff
@@ -151,90 +151,22 @@ public String getVersion() {
 
     }
 
-    private static class TargetRef implements TargetDefinition.TargetReferenceLocation {
-
-        private String uri;
-
-        public TargetRef(String uri) {
-            this.uri = uri;
-        }
-
+	private record TargetReference(String getUri) implements TargetDefinition.TargetReferenceLocation {
         @Override
         public String getTypeDescription() {
             return "Target";
         }
-
-        @Override
-        public String getUri() {
-            return uri;
-        }
-
     }
 
-	private static final class OSGIRepositoryLocation implements TargetDefinition.RepositoryLocation {
-
-		private String uri;
-		private Collection<Requirement> requirements;
-
-		public OSGIRepositoryLocation(String uri, Collection<Requirement> requirements) {
-			this.uri = uri;
-			this.requirements = requirements;
-		}
-
-		@Override
-		public String getUri() {
-			return uri;
-		}
-
-		@Override
-		public Collection<Requirement> getRequirements() {
-			return requirements;
-		}
-
+	private record OSGIRepositoryLocation(String getUri, Collection<Requirement> getRequirements)
+			implements TargetDefinition.RepositoryLocation {
 	}
 
-    private static class MavenLocation implements TargetDefinition.MavenGAVLocation {
-
-        private final Collection<String> includeDependencyScopes;
-        private final MissingManifestStrategy manifestStrategy;
-        private final boolean includeSource;
-        private final Collection<BNDInstructions> instructions;
-        private final Collection<MavenDependency> roots;
-        private final DependencyDepth dependencyDepth;
-        private final Collection<MavenArtifactRepositoryReference> repositoryReferences;
-        private final Element featureTemplate;
-		private String label;
-
-        public MavenLocation(Collection<MavenDependency> roots, Collection<String> includeDependencyScopes,
-                MissingManifestStrategy manifestStrategy, boolean includeSource,
-                Collection<BNDInstructions> instructions, DependencyDepth dependencyDepth,
-				Collection<MavenArtifactRepositoryReference> repositoryReferences, Element featureTemplate,
-				String label) {
-            this.roots = roots;
-            this.includeDependencyScopes = includeDependencyScopes;
-            this.manifestStrategy = manifestStrategy;
-            this.includeSource = includeSource;
-            this.instructions = instructions;
-            this.dependencyDepth = dependencyDepth;
-            this.repositoryReferences = repositoryReferences;
-			this.label = label;
-            this.featureTemplate = featureTemplate == null ? null : (Element) featureTemplate.cloneNode(true);
```

Comment:
The element is no longer cloned as far as I can see.

Reply:
Indeed. Thanks.
Fixed it.